conf/modules.config

```diff
@@ -300,14 +353,74 @@ process {
         ]
     }
 
-    withName: SAMTOOLS_FAIDX {
-        ext.prefix = { "${meta.id}_${meta.read_type}" }
-        tag        = { "${meta.id}_${meta.read_type}" }
+    withName: SAMTOOLS_FAIDX_HIC_MAPPING {
+        ext.prefix = { "${meta.id}_${meta.assembly_type}_${meta.haplotype}" }
+        tag        = { "${meta.id}_${meta.assembly_type}_${meta.haplotype}" }
+    }
+
+    withName: SAMTOOLS_FAIDX_POLISHING {
+        ext.prefix = { "${meta.id}_${meta.assembly_type}" }
+        tag        = { "${meta.id}_${meta.assembly_type}" }
+    }
+
+    withName: SAMTOOLS_FLAGSTAT {
+        ext.prefix = { "${meta.id}_${meta.assembly_type}_${meta.haplotype}" }
+        ext.prefix = { "${meta.id}_${meta.assembly_type}_${meta.haplotype}" }
+        publishDir = [
+            path: { "${params.outdir}/${myfunctions.assemblyDir(meta, workflow.start)}/scaffolding_${meta.haplotype}" },
+            mode: params.publish_dir_mode,
+            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+        ]
+    }
+
+    withName: SAMTOOLS_INDEX_HIC_CRAM {
+        ext.prefix = { "${input.getName()}" }
+        tag        = { "${input.getName()}" }
+    }
+
+    withName: SAMTOOLS_INDEX_HIC_MAP {
+        ext.prefix = { "${meta.id}_${meta.assembly_type}_${meta.haplotype}" }
+        tag        = { "${meta.id}_${meta.assembly_type}_${meta.haplotype}" }
+    }
+
+    withName: SAMTOOLS_IDXSTATS {
+        ext.prefix = { "${meta.id}_${meta.assembly_type}_${meta.haplotype}" }
+        tag        = { "${meta.id}_${meta.assembly_type}_${meta.haplotype}" }
+        publishDir = [
+            path: { "${params.outdir}/${myfunctions.assemblyDir(meta, workflow.start)}/scaffolding_${meta.haplotype}" },
+            mode: params.publish_dir_mode,
+            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+        ]
+    }
+
+    withName: SAMTOOLS_MARKDUP_HIC_MAPPING {
+        ext.prefix = { "${meta.id}_${meta.assembly_type}_${meta.haplotype}" }
+        tag        = { "${meta.id}_${meta.assembly_type}_${meta.haplotype}" }
+        publishDir = [
+            path: { "${params.outdir}/${myfunctions.assemblyDir(meta, workflow.start)}/scaffolding_${meta.haplotype}" },
+            mode: params.publish_dir_mode,
+            saveAs: { filename -> filename.contains('*.bam') ? "merge.mkdup.bam" : null }
+        ]
+    }
+
+    withName: 'SAMTOOLS_MERGE_HIC_MAPPING' {
+        ext.prefix = { "${meta.id}_${meta.assembly_type}_${meta.haplotype}_merged" }
+        tag        = { "${meta.id}_${meta.assembly_type}_${meta.haplotype}"        }
+    }
+
+    withName: SAMTOOLS_STATS {
+        ext.prefix = { "${meta.id}_${meta.assembly_type}_${meta.haplotype}" }
+        tag        = { "${meta.id}_${meta.assembly_type}_${meta.haplotype}" }
```

Comment:
At this point, could you not consider changing the value of `meta.id` to `${meta.id}_${meta.assembly_type}_${meta.haplotype}` somewhere in a sub-workflow, so that the default value of `ext.prefix` and `tag` just work ?

Reply:
I think because I’m doing a lot of joining on the meta object, that would get unwieldy fast. So this is wordy, but reliable